### PR TITLE
Expand SSRF support in IAST to java.net.http.HttpClient

### PR DESCRIPTION
**Pull request by Mariovido**
# What Does This Do
Add support for the `java-net` client library to detect **SSRF**. This is done by detecting the vulnerability using the `HttpClientDecorator`.

The new `java.net.http.HttpClient` methods that will be supported are:
- `send(HttpRequest, HttpResponse.BodyHandler<T>)`
- `sendAsync(HttpRequest, HttpResponse.BodyHandler<T>)`
- `sendAsync(HttpRequest, HttpResponse.BodyHandler<T>, HttpResponse.PushPromiseHandler<T>)`

# Motivation
With this change we want to expand the support for **SSRF** in the different clients supported by the `HttpClientDecorator`.

# Additional Notes
Apart from detecting the vulnerability, a new _smoke-test_, which uses **Java 11**, has been created.

# Contributor Checklist

- [x] Format the title [according the contribution guidelines](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#title-format)
- [x] Assign the `type:` and (`comp:` or `inst:`) labels in addition to [any usefull labels](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#labels)
- [ ] Squash your commits prior merging or merge using GitHub's [Squash and merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits)
- [x] Don't use `close`, `fix` or any [linking keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) when referencing an issue.  
  Use `solves` instead, and assign the PR [milestone](https://github.com/DataDog/dd-trace-java/milestones) to the issue
- [x] Update the [public documentation](https://docs.datadoghq.com/tracing/trace_collection/library_config/java/) in case of new configuration flag or behavior

Jira ticket: [APPSEC-55633]

[APPSEC-55633]: https://datadoghq.atlassian.net/browse/APPSEC-55633?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dd-smoke-tests/iast-util/src/testFixtures/groovy/datadog/smoketest/AbstractIastSpringBootTest.groovy [hunk 1]
```diff
@@ -753,12 +753,12 @@ abstract class AbstractIastSpringBootTest extends AbstractIastServerSmokeTest {
     }
 
     where:
-    path                  | parameter | value                     | method               | protocolSecure
-    "apache-httpclient4"  | "url"     | "https://dd.datad0g.com/" | "apacheHttpClient4"  | false
-    "apache-httpclient4"  | "host"    | "dd.datad0g.com"          | "apacheHttpClient4"  | false
-    "commons-httpclient2" | "url"     | "https://dd.datad0g.com/" | "commonsHttpClient2" | false
-    "okHttp2"             | "url"     | "https://dd.datad0g.com/" | "okHttp2"            | false
-    "okHttp3"             | "url"     | "https://dd.datad0g.com/" | "okHttp3"            | false
+    path                  | parameter | value                     | protocolSecure
```
Comment (jandro996) [inline dd-smoke-tests/iast-util/src/testFixtures/groovy/datadog/smoketest/AbstractIastSpringBootTest.groovy:756]: 👍

File: dd-smoke-tests/iast-util/iast-util-11/src/testFixtures/groovy/datadog/smoketest/AbstractIast11SpringBootTest.groovy [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      if (parameter == 'url') {
+        return parts.size() == 1
+        && parts[0].value == value && parts[0].source.origin == 'http.request.parameter' && parts[0].source.name == parameter
+      } else {
+        throw new IllegalArgumentException("Parameter $parameter not supported")
+      }
+    }
+
+    where:
+    path       | parameter | value                     | async   | promise
```
Comment (jandro996) [inline dd-smoke-tests/iast-util/iast-util-11/src/testFixtures/groovy/datadog/smoketest/AbstractIast11SpringBootTest.groovy:63]: No sure if you did that thinking in future cases but seems that path, parameter, and value are always the same so they can be outside the where clause simplifying it.

**Review by jandro996 — APPROVED** [review]
LGTM!

File: dd-smoke-tests/springboot-java-11/src/main/java/datadog/smoketest/springboot/SpringbootApplication.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package datadog.smoketest.springboot;
+
+import java.lang.management.ManagementFactory;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
```
Comment (jandro996) [inline dd-smoke-tests/springboot-java-11/src/main/java/datadog/smoketest/springboot/SpringbootApplication.java:7]: @ComponentScan is not necessary to register the SSRFController?

File: dd-smoke-tests/iast-util/iast-util-11/src/testFixtures/groovy/datadog/smoketest/AbstractIast11SpringBootTest.groovy [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    .add(parameter, value)
+    .add("async", async)
+    .add("promise", promise).build()
+    final request = new Request.Builder().url(url).post(body).build()
+
+    when:
+    client.newCall(request).execute()
+
+    then:
+    hasVulnerability { vul ->
```
Comment (jandro996) [inline dd-smoke-tests/iast-util/iast-util-11/src/testFixtures/groovy/datadog/smoketest/AbstractIast11SpringBootTest.groovy:49]: Not sure if you need so many complexity for this tests

```suggestion
    hasVulnerability { vul -> vul.type == 'SSRF' }
```

**Review by ValentinZakharov — APPROVED** [review]

File: dd-smoke-tests/iast-util/iast-util-11/src/testFixtures/groovy/datadog/smoketest/AbstractIast11SpringBootTest.groovy (hunk 2 above)
Comment (Mariovido) [reply dd-smoke-tests/iast-util/iast-util-11/src/testFixtures/groovy/datadog/smoketest/AbstractIast11SpringBootTest.groovy:63]: I prefer to leave it like this due to the fact that some libraries made calls with `host` and no `url`. It is true that now only uses url but probably in the future there will be some with `host`.

File: dd-smoke-tests/iast-util/iast-util-11/src/testFixtures/groovy/datadog/smoketest/AbstractIast11SpringBootTest.groovy (hunk 4 above)
Comment (Mariovido) [reply dd-smoke-tests/iast-util/iast-util-11/src/testFixtures/groovy/datadog/smoketest/AbstractIast11SpringBootTest.groovy:49]: It's true that is not needed, but I prefer to do it like this to make the tests similar to the ones that we have inside `AbstractIastSpringBootTest`

**Review by smola — APPROVED** [review]

**Pull request merged**